drivers/led/adp8866.c

```diff
@@ -0,0 +1,345 @@
+	 * nibble values
+	 *  0001 = 0.05 sec.
+		0010 = 0.10 sec.
+		0011 = 0.15 sec.
+		0100 = 0.20 sec.
+		0101 = 0.25 sec.
+		0110 = 0.30 sec.
+		0111 = 0.35 sec.
+		1000 = 0.40 sec.
+		1001 = 0.45 sec.
+		1010 = 0.50 sec.
+		1011 = 0.75 sec.
+		1100 = 1.00 sec.
+		1101 = 1.25 sec.
+		1110 = 1.50 sec.
+		1111 = 1.75 s
+	 */
+	if (i2c_reg_write_byte(data->i2c, APD8866_I2C_ADDR,
+			APD8866_REG_ISCF, 0x42)) {
+
+		LOG_ERR("Setting ICS fade rate failed.");
+		return -EIO;
+	}
+
+	LOG_INF("ICS fade rate set");
+
+
+	data->mdcr = 0x65;
+
+	if (i2c_reg_write_byte(data->i2c, APD8866_I2C_ADDR,
+			APD8866_REG_MDCR,data->mdcr)) {
+		LOG_ERR("Setting backlight fade rate failed.");
+		return -EIO;
+	}
+
+
+	LOG_INF("backlight fade rate set");
+
+	if (i2c_reg_write_byte(data->i2c, APD8866_I2C_ADDR,
+			APD8866_REG_ISCC1, 0x6)) {
+		LOG_ERR("Setting ISCC1 failed.");
+		return -EIO;
+	}
+
+	LOG_INF("ISCC1 set");
+	if (i2c_reg_write_byte(data->i2c, APD8866_I2C_ADDR,
+			APD8866_REG_ISCC2, 0xff)) {
+		LOG_ERR("Setting ISCC2 (Independent Sink Current Control) failed.");
+		return -EIO;
+	}
+
+	LOG_DBG("ISCC2 set");
+
+	// set ISC for led 9 as well
+	if (i2c_reg_write_byte(data->i2c, APD8866_I2C_ADDR, APD8866_REG_CFGR, 0x14)) {
+
+		LOG_ERR("Setting ISC for LED9 failed.");
+		// return 0;
+		return -EIO;
+	}
+
+	// dunno if this is required...
+	i2c_reg_write_byte(data->i2c, APD8866_I2C_ADDR,
+			APD8866_REG_PWR_SEL1, 0);
+	i2c_reg_write_byte(data->i2c, APD8866_I2C_ADDR,
+			APD8866_REG_PWR_SEL2, 0);
+
+
+
+	if (i2c_reg_read_byte(data->i2c, APD8866_I2C_ADDR,
+			APD8866_REG_INT_STAT, &(data->int_stat))) {
+
+		LOG_ERR("Reading ADP8866 chip failed.");
+		return -EIO;
+	}
+	LOG_INF("GOT stat %i", data->int_stat);
+	if (data->int_stat & (0x1C)) {
+
+		LOG_ERR("Error condition reported--check int_stat");
+		return -EIO;
+	}
+
+
+	LOG_DBG("LED driver setup done");
+	data->setup_is_done = 1;
+	return 0;
+}
+
+
+static int adp8866_led_init(const struct device *dev)
+{
+
+	LOG_INF("LED INIT CALLED");
+	struct adp8866_data *data = dev->data;
+	data->setup_is_done = 0;
+	/* struct led_data *dev_data = &data->dev_data; */
+	unsigned char tmp;
+
+		if (! data->i2c) {
+
+			data->i2c = device_get_binding(DT_INST_BUS_LABEL(0));
+			if (data->i2c == NULL) {
+				LOG_ERR("Failed to get I2C device");
+				return -EINVAL;
+			}
+			LOG_INF("Got I2C device, addr is %i", APD8866_I2C_ADDR);
+		}
+
+		if (i2c_reg_read_byte(data->i2c, APD8866_I2C_ADDR,
+				APD8866_REG_MFDVID, &(tmp))) {
+
+			LOG_ERR("Reading ADP8866 chip failed.");
+			return -EIO;
+		}
+
+		if (! tmp) {
+			LOG_ERR("Could not get device ID");
+			return -EIO;
+		}
+		LOG_DBG("DEV ID is %i", tmp);
+		data->deviceID = tmp & 0x0f;
+		data->manufacturerID = tmp >> 4;
+
+	return 0;
+
+}
+
+
+static int adp8866_led_blink(const struct device *dev, uint32_t led,
+			    uint32_t delay_on, uint32_t delay_off)
+{
+	struct adp8866_data *data = dev->data;
+	/*
+	 * TODO: FIXME -- unimplemented
+	 */
+
+	return -EIO;
+}
```

Comment:
Please consider using a "draft" PR status until it is ready to review.

Reply:
optional, got rid of it and cleaned up a bit.  hopefully this is now in a satisfying condition.